### DISCUSSION TITLE
OpenModelica and translations

### DISCUSSION THREAD

**Discussion by s-celles** [OpenModelica/OpenModelica#11222] [Q&A]
Hello,

OpenModelica is partially translated from english to french (and many other languages).
I'd like to contribute to provide more translated strings.

I saw that https://github.com/OpenModelica/OpenModelica/tree/master/OMEdit/OMEditLIB/Resources/nls directory contains translation strings but I don't really want to install Qt SDK just for that.

Files in this directory are 3 years old... not very active translation activity there !

Isn't there a way to move to online translating platform such as [Transifex](https://www.transifex.com/), [Crowdin](https://crowdin.com/), [Weblate](https://weblate.org/en/), [POEditor](https://poeditor.com/) ... to make translating contributions easier.

Kind regards

**Comment by s-celles** [comment]
I tried to fork the repository and followed https://github.com/OpenModelica/OpenModelica#working-with-the-repository

unfortunately I'm running Windows and couldn't find why last command is failing:

  
    C:\Users\scell\github\scls19fr> $MY_FORK="scls19fr"
    C:\Users\scell\github\scls19fr> git clone --recurse-submodules https://openmodelica.org/git-readonly/OpenModelica.git
    Cloning into 'OpenModelica'...
    remote: Enumerating objects: 307795, done.
    remote: Counting objects: 100% (41272/41272), done.
    remote: Compressing objects: 100% (9017/9017), done.
    remote: Total 307795 (delta 38903), reused 32881 (delta 32219), pack-reused 266523
    Receiving objects: 100% (307795/307795), 234.02 MiB | 5.83 MiB/s, done.
    
    Resolving deltas: 100% (240821/240821), done.
    Updating files: 100% (15185/15185), done.
    Submodule 'OMCompiler/3rdParty' (https://openmodelica.org/git-readonly/OMCompiler-3rdParty) registered for path 'OMCompiler/3rdParty'
    Submodule 'OMOptim' (https://openmodelica.org/git-readonly/OMOptim.git) registered for path 'OMOptim'
    Submodule 'OMSens' (https://openmodelica.org/git-readonly/OMSens.git) registered for path 'OMSens'
    Submodule 'OMSens_Qt' (https://openmodelica.org/git-readonly/OMSens_Qt.git) registered for path 'OMSens_Qt'
    Submodule 'OMSimulator' (https://openmodelica.org/git-readonly/OMSimulator.git) registered for path 'OMSimulator'
    Cloning into 'C:/Users/scell/github/scls19fr/OpenModelica/OMCompiler/3rdParty'...
    remote: Enumerating objects: 48749, done.
    remote: Counting objects: 100% (48749/48749), done.
    remote: Compressing objects: 100% (19022/19022), done.
    remote: Total 48749 (delta 30827), reused 45672 (delta 28765), pack-reused 0
    Receiving objects: 100% (48749/48749), 79.61 MiB | 5.03 MiB/s, done.
    Cloning into 'C:/Users/scell/github/scls19fr/OpenModelica/OMOptim'...
    remote: Counting objects: 100% (10164/10164), done.
    remote: Total 10164 (delta 7576), reused 9669 (delta 7200), pack-reused 0
    Receiving objects: 100% (10164/10164), 8.50 MiB | 4.86 MiB/s, done.
    Resolving deltas: 100% (7576/7576), done.
    Cloning into 'C:/Users/scell/github/scls19fr/OpenModelica/OMSens'...
    remote: Enumerating objects: 3310, done.
    remote: Counting objects: 100% (3310/3310), done.
    remote: Compressing objects: 100% (1180/1180), done.
    remote: Total 3310 (delta 2107), reused 3243 (delta 2075), pack-reused 0
    Receiving objects: 100% (3310/3310), 7.95 MiB | 3.20 MiB/s, done.
    Resolving deltas: 100% (2107/2107), done.
    Cloning into 'C:/Users/scell/github/scls19fr/OpenModelica/OMSens_Qt'...
    remote: Enumerating objects: 1823, done.
    remote: Counting objects: 100% (1823/1823), done.
    remote: Compressing objects: 100% (679/679), done.
    remote: Total 1823 (delta 1341), reused 1522 (delta 1139), pack-reused 0
    Receiving objects: 100% (1823/1823), 14.08 MiB | 2.06 MiB/s, done.
    Resolving deltas: 100% (1341/1341), done.
    Cloning into 'C:/Users/scell/github/scls19fr/OpenModelica/OMSimulator'...
    remote: Enumerating objects: 15050, done.
    remote: Counting objects: 100% (15050/15050), done.
    remote: Compressing objects: 100% (6738/6738), done.
    remote: Total 15050 (delta 11219), reused 11467 (delta 8112), pack-reused 0
    Receiving objects: 100% (15050/15050), 67.11 MiB | 3.74 MiB/s, done.
    Resolving deltas: 100% (11219/11219), done.
    Submodule path 'OMCompiler/3rdParty': checked out '449e88fbe5fb25f04b4520b5b7ee472de27af7d4'
    Submodule path 'OMOptim': checked out '55102095970c73a4176820538d19aaf408515da9'
    Submodule 'common' (https://openmodelica.org/git-readonly/OpenModelica-common.git) registered for path 'OMOptim/common'
    Cloning into 'C:/Users/scell/github/scls19fr/OpenModelica/OMOptim/common'...
    remote: Enumerating objects: 141, done.
    remote: Counting objects: 100% (141/141), done.
    remote: Compressing objects: 100% (120/120), done.
    remote: Total 141 (delta 84), reused 46 (delta 21), pack-reused 0
    Receiving objects: 100% (141/141), 48.00 KiB | 1.12 MiB/s, done.
    Resolving deltas: 100% (84/84), done.
    Submodule path 'OMOptim/common': checked out '08a01802db5ba5edb540383c46718b89ff229ef2'
    Submodule path 'OMSens': checked out '8216eaefbb4ea2ff0c7a17a3487dcc2a8a17667d'
    Submodule path 'OMSens_Qt': checked out 'ff2ade18e7058f611ddfd575ccbd24f388615d3f'
    Submodule 'common' (https://github.com/OpenModelica/OpenModelica-common) registered for path 'OMSens_Qt/common'
    Cloning into 'C:/Users/scell/github/scls19fr/OpenModelica/OMSens_Qt/common'...
    remote: Enumerating objects: 141, done.
    remote: Counting objects: 100% (1/1), done.
    remote: Total 141 (delta 0), reused 0 (delta 0), pack-reused 140
    Receiving objects: 100% (141/141), 50.51 KiB | 10.10 MiB/s, done.
    Resolving deltas: 100% (82/82), done.
    Submodule path 'OMSens_Qt/common': checked out '08a01802db5ba5edb540383c46718b89ff229ef2'
    Submodule path 'OMSimulator': checked out '634b0193a3eb14e75622a8e720bf0fb6dab77487'
    Submodule '3rdParty' (https://github.com/OpenModelica/OMSimulator-3rdParty.git) registered for path 'OMSimulator/3rdParty'
    Cloning into 'C:/Users/scell/github/scls19fr/OpenModelica/OMSimulator/3rdParty'...
    remote: Enumerating objects: 11623, done.
    remote: Counting objects: 100% (463/463), done.
    remote: Compressing objects: 100% (208/208), done.
    remote: Total 11623 (delta 308), reused 255 (delta 255), pack-reused 11160
    Receiving objects: 100% (11623/11623), 27.89 MiB | 5.08 MiB/s, done.
    Resolving deltas: 100% (4782/4782), done.
    Submodule path 'OMSimulator/3rdParty': checked out 'b4c3aa4a16a5920e88096780498cc4af157f3c05'
    C:\Users\scell\github\scls19fr> cd OpenModelica
    C:\Users\scell\github\scls19fr\OpenModelica [master ≡]> git remote set-url --push origin git@github.com:$MY_FORK/OpenModelica.git
    C:\Users\scell\github\scls19fr\OpenModelica [master ≡]> git submodule foreach --recursive 'git remote set-url --push origin `git config --get remote.origin.url | sed s,^.*/,git@github.com:'$MY_FORK'/,`'
    Entering 'OMCompiler/3rdParty'
    git remote set-url --push origin `git config --get remote.origin.url | sed s,^.*/,git@github.com:: -c: line 1: unexpected EOF while looking for matching ``'
    fatal: run_command returned non-zero status for OMCompiler/3rdParty

I'm using Windows 10 with PowerShell

```
> $PSVersionTable.PSVersion

Major  Minor  Build  Revision
-----  -----  -----  --------
5      1      19041  3031
```

I'm using the following git version

```
> git --version
git version 2.41.0.windows.1
```

Any idea?


PS : Using [Cygwin](https://www.cygwin.com/) I endup doing

```
scell@DESKTOP-PHOENIX /cygdrive/c/Users/scell/github/scls19fr/OpenModelica
$ MY_FORK=scls19fr

scell@DESKTOP-PHOENIX /cygdrive/c/Users/scell/github/scls19fr/OpenModelica
$ git submodule foreach --recursive 'git remote set-url --push origin `git config --get remote.origin.url | sed s,^.*/,git@github.com:'$MY_FORK'/,`'
Entering 'OMCompiler/3rdParty'
Entering 'OMOptim'
Entering 'OMOptim/common'
Entering 'OMSens'
Entering 'OMSens_Qt'
Entering 'OMSens_Qt/common'
Entering 'OMSimulator'
Entering 'OMSimulator/3rdParty'
```

**Comment by mahge** [comment]
I am not sure what the issue is but personally I recommend just adding a new remote instead of modifying the the push url for the origin remote. 

```sh
git clone --recurse-submodules https://openmodelica.org/git-readonly/OpenModelica.git
cd  OpenModelica

# Add a new remote for the root repository.
git remote add scell https://github.com/scls19fr/OpenModelica.git

# Add a new remote for the submodule repositories you plan to update.
cd OMCompiler/3rdParty
git remote add scell https://github.com/scls19fr/OMCompiler-3rdParty
...
```

You operate the same way but you now push to your remote explicitly. 

```sh
git checkout -b my_fixes
git add ...
git commit ...

# Push with explicit remote.
git push scell 
```

Then open a pull request from it. Of course this is a personal preference but I find this easier, more conventional and less confusing.

**Comment by adeas31** [comment]
The translations stuff is mostly volunteer work. Unfortunately, you need to install Qt to contribute.

We might think about online platforms if they are free and provide support for Qt.

**Reply by s-celles** [reply]
[Codeberg](https://codeberg.org/) provides a free [Weblate](https://weblate.org/) instance https://translate.codeberg.org/

**Reply by comradekingu** [reply]
It is under a CoC. Use the Libre plan at https://hosted.weblate.org/hosting/ instead :)

**Reply by s-celles** [reply]
What is CoC meaning?

**Comment by lucas-mior** [comment]
I don't mean to be rude, but translating openmodelica is a waste of time. I doubt someone who doesnt understand english could do anything useful with OM, considering how all over the place the documentation and tools are. One can't even know for sure wheter a specific OM tool has been abandoned or not (because there are lots of them, and they interact in confusing ways).

**Reply by casella** [reply]
@lucas-mior I also believe that nowadays most people doing technical work have at least a basic grasp of English, like it or not, and that's why we are not really putting any resources on that.

Regarding the documentation, we are trying hard to collect all the relevant information and pointers for users in the [User's Guide](https://openmodelica.org/doc/OpenModelicaUsersGuide/latest/index.html). If you see any obvious issues with it, please open tickets, and if you want to actively contribute to the writing of the documentation, that would be great 😃
